src/Sylius/Component/Taxation/Checker/TaxRateDateEligibilityChecker.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Taxation\Checker;
+
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
+use Sylius\Component\Taxation\Model\TaxRateInterface;
+
+final class TaxRateDateEligibilityChecker implements TaxRateDateEligibilityCheckerInterface
+{
+    public function __construct(
+        protected DateTimeProviderInterface $calendar,
+    ) {
+    }
+
+    public function isEligible(TaxRateInterface $taxRate): bool
+    {
+        $date = $this->calendar->now();
+        $startDate = $taxRate->getStartDate();
+        $endDate = $taxRate->getEndDate();
+
+        if (null === $endDate) {
+            return $startDate <= $date;
+        }
+
+        return $startDate <= $date && $endDate >= $date;
```

Comment:
This can be a one-liner.
It will also avoid comparing null to DateTime when `$startDate` is null.
```suggestion
        return  (null === $startDate || $startDate <= $date) &&  (null === $endDate || $endDate >= $date);
```